src/utils/management/CustomState.ts

```diff
@@ -0,0 +1,34 @@
+export class CustomState<T> {
+  private error: string;
+  private pending: boolean;
+  private data: T;
+  private readonly preparedData: T;
+
+  constructor(preparedData: T) {
+    this.preparedData = preparedData;
+  }
+
+  public init(): CustomState<T> {
```

Comment:
Avoid using the public keyword. In programming, avoid practices that generate additional code - due to the extra workload of keeping an eye on such conventions. Public is default so can be avoided - except constructor.
In constructor public is shorthand syntax.